### PR TITLE
Respect current byteOffset in subarray

### PR DESCRIPTION
**Pull request by LinusU**

### CODE REVIEW

**Comment by dcodeIO** [comment]
Great, thanks! :)

File: std/assembly/internal/typedarray.ts [hunk 1]
```diff
@@ -90,7 +90,7 @@ export abstract class TypedArray<T,V> {
     else end = max(min(end, length), begin);
     var slice = memory.allocate(offsetof<this>());
     store<usize>(slice, this.buffer, offsetof<this>("buffer"));
-    store<i32>(slice, begin << alignof<T>(), offsetof<this>("byteOffset"));
+    store<i32>(slice, this.byteOffset + (begin << alignof<T>()), offsetof<this>("byteOffset"));
     store<i32>(slice, (end - begin) << alignof<T>(), offsetof<this>("byteLength"));
```
Comment (MaxGraey) [inline std/assembly/internal/typedarray.ts]: Is not this should be:
```ts
(end - begin - this.byteOffset) << alignof<T>()
```
?

Comment (dcodeIO) [reply std/assembly/internal/typedarray.ts]: Isn't `byteLength` independent of `byteOffset`? Like really just the length of the data in bytes?

Comment (MaxGraey) [reply std/assembly/internal/typedarray.ts]: Yeah, right!

**Pull request merged**